test/verify/check-machines

```diff
@@ -1110,15 +1115,41 @@ class TestMachines(MachineCase):
             b.wait_val(selector, value)
 
     class CreateVmRunner:
-        def __init__(self, env):
-            self.browser = env.browser
-            self.machine = env.machine
+        def __init__(self, test_obj):
+            self.browser = test_obj.browser
+            self.machine = test_obj.machine
+            self.assertTrue = test_obj.assertTrue
 
             self.machine.execute("touch {0}".format(TestMachines.TestCreateConfig.NOVELL_MOCKUP_ISO_PATH))
 
         def destroy(self):
             self.machine.execute("rm -f {0}".format(TestMachines.TestCreateConfig.NOVELL_MOCKUP_ISO_PATH))
 
+        @staticmethod
+        def assertVmStates(test_obj, name, before, wanted, after):
+            b = test_obj.browser
+            selector = "#vm-{0}-state".format(name)
+
+            def waitForWanted(accepted, wanted):
+                b.wait_present(selector)
+                try:
+                    text = b.text(selector)
+                except:  # if selector disappears
+                    return False
+
+                if (accepted and accepted in text) or text == 'in transition':
+                    return False
+                elif wanted in text:
+                    return True
+                else:
+                    raise Exception("invalid vm state")
+
+            with b.wait_timeout(2):
```

Comment:
Our CI is often rather loaded, so 2 seconds is a rather optimistic deadline. But I'm okay with this for now, but we might need to bump it if we see this flaking.